src/Repository/DBALEventRepository.php

```diff
@@ -48,14 +61,16 @@ public function appendTo(string $streamName, AggregateRootId $aggregateRootId, a
         ) VALUES '.\implode(',', \array_fill(0, $eventCount, '(?, ?, ?, ?, ?, ?, ?)'));
 
         $data = [];
-        foreach ($events as $event) {
-            $eventData = $this->eventConverter->convertToArray($event);
+        $metadata = $eventStream->getMetadata();
+
+        foreach ($eventStream as $event) {
+            $eventData = $this->eventConverter->convertToArray($event, $metadata);
 
             $data[] = $eventData['event_id'];
             $data[] = \get_class($event);
-            $data[] = (string) $aggregateRootId;
-            $data[] = $eventData['payload'];
-            $data[] = $eventData['metadata'];
+            $data[] = $eventData['aggregate_id'];
+            $data[] = $this->jsonEncoder->encode($eventData['payload']);
```

Comment:
Why has the responsibility for (json) encoding moved from the event convertor logic to the repository?

Reply:
It was necessary to decorate the `ArraySerializableEventConverter` with `FieldEncryptingConverter`. Within the encryptor an payload array is needed to allow encryption per field. If it is already converted to json, it would have to be decoded/encoded again.
It also makes more sense. The json encoding is a requirement for the `DBALRepository`, there could be a repository implementation that can handle arrays.